src/Mod/MeshPart/App/AppMeshPartPy.cpp

```diff
@@ -509,7 +512,15 @@ class Module : public Py::ExtensionModule<Module>
                 }
                 mesher.setColors(colors);
             }
-            return Py::asObject(new Mesh::MeshPy(mesher.createMesh()));
+            Mesh::MeshObject* mesh;
+            if (Base::asBoolean(releaseGIL)) {
+                Py_BEGIN_ALLOW_THREADS
+                mesh = mesher.createMesh();
+                Py_END_ALLOW_THREADS
+            } else {
+                mesh = mesher.createMesh();
+            }
+            return Py::asObject(new Mesh::MeshPy(mesh));

```

Comment:
OK, I've updated the code.

Reply:
I don't know your specific use case, but note that it is possible to parallelize the creation of the mesh (`BRepMesh_IncrementalMesh` supports parallelization).

A comment about using git:
It is recommended to make changes on a local branch different than your local main and then use `git rebase` (not merge) to apply the changes on top the `FreeCAD:main` branch.
You are currently submitting a pull request with several commits already present in `FreeCAD:main` because you have updated your local branch using merge.
Also, in this case, it is preferable to squash the commits you are trying to send, so that only the final version that uses `Base::PyGILStateRelease` remains. The first commit that uses `Py_*_ALLOW_THREADS` is unnecessary.